app/views/tasks/show.html.slim

```diff
@@ -478,13 +478,21 @@
             = button_tag class: 'btn btn-light dropdown-toggle', 'data-bs-toggle': 'dropdown' do
               = t('.button.add_to_collection')
             ul.scrollable.dropdown-menu[role="menu"]
-              - if !current_user.collections.empty?
-                - current_user.collections.reverse.each do |collection|
-                  li
-                    = link_to collection.title, add_to_collection_task_path(collection: collection), method: 'post', class: 'dropdown-item'
-              - else
+              - current_user.collections.order(created_at: :desc).each do |collection|
                 li
-                  = link_to t('.button.add_collection'), new_collection_path(current_user)
+                  - if collection.tasks.include?(@task)
+                    = link_to "<span class=\"collection-checkmark\">✓</span>#{collection.title}".html_safe, remove_task_ajax_collection_path(collection, task: @task, return_to_task: true), method: :patch, class: 'dropdown-item', data: { confirm: t('.remove_task_from_collection_warning') }
```

Comment:
This still looks a bit sketchy to me.

Probably, I'd go for `link_to ... do`, which is better readable and meant for such use cases.

Reply:
You are right, this looks way better now

Reply:
Nice, thanks for taking another look here. Personally, I think the `link_to ... do` solution now implemented is much cleaner and better to maintain 💪